### ISSUE TITLE
Installation issue: WRF

### ISSUE THREAD

**Issue by christian-heusel** [LLNL/spack#20581]
#### Error 1

```console
$ spack install wrf
[...]
==> wrf: Executing phase: 'build'
==> Error: TypeError: int() can't convert non-string with explicit base

/home/chris/Documents/shared_projects/spack/var/spack/repos/builtin/packages/wrf/package.py:292, in run_compile_script:
        289        csh = Executable(csh_bin)
        290
        291        # num of compile jobs capped at 20 in wrf
  >>    292        num_jobs = str(min(int(make_jobs, 10)))
        293
        294        # Now run the compile script and track the output to check for
        295        # failure/success We need to do this because upstream use `make -i -k`

See build log for details:
[...]
```

#### Error 2

```
==========================================================================
build started:   Mon Dec 28 16:18:26 CET 2020
build completed: Mon Dec 28 16:29:26 CET 2020

--->                  Executables successfully built                  <---

-rwxr-xr-x 1 hd_eq192 hd_hd 45736584 Dec 28 16:29 main/ndown.exe
-rwxr-xr-x 1 hd_eq192 hd_hd 45576712 Dec 28 16:29 main/real.exe
-rwxr-xr-x 1 hd_eq192 hd_hd 44987304 Dec 28 16:29 main/tc.exe
-rwxr-xr-x 1 hd_eq192 hd_hd 51173080 Dec 28 16:28 main/wrf.exe

==========================================================================

==> Error: TypeError: argument of type 'NoneType' is not iterable

/beegfs/home/myname/spack/var/spack/repos/builtin/packages/wrf/package.py:305, in run_compile_script:
        302            self.spec.variants["compile_type"].value,
        303        )
        304
  >>    305        if "Executables successfully built" in result_buf:
        306            return True
        307
        308        return False
```

#### Error 1

I noticed that the error was most likely caused by 42008e532f28c41d8e134052e1a109c65a8440f9 in this changeset:
```diff
diff --git a/var/spack/repos/builtin/packages/wrf/package.py b/var/spack/repos/builtin/packages/wrf/package.py
--- a/var/spack/repos/builtin/packages/wrf/package.py
+++ b/var/spack/repos/builtin/packages/wrf/package.py
@@ -138,5 +287,6 @@
-    def build(self, spec, prefix):
-        csh = which('csh')
+    def run_compile_script(self):
+        csh_bin = self.spec["tcsh"].prefix.bin.csh
+        csh = Executable(csh_bin)
+
         # num of compile jobs capped at 20 in wrf
-        csh('./compile', '-j', str(min(int(make_jobs), 20)),
-            spec.variants['compile_type'].value)
+        num_jobs = str(min(int(make_jobs, 10)))

```
The comment notes that the build should have a maximum of 20 build jobs, is that or 10 correct?
I think there just is an error in the parentheses, see https://github.com/christian-heusel/spack/commit/8c35486037dfbe1b25e9e779c377dddc02f1d3f9

#### Error 2

I am still investigating this and will post updates if I find anything 😊 

### Information on your system

* **Spack:** 0.16.0
* **Python:** 3.9.1
* **Platform:** linux-debianunstable-zen2

### Additional information

I dont think that a build report is necessary here, feel free to correct me if it yields any useful information we do not have yet here! 😊 

@MichaelLaufer  @ptooley

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
Can you see if this is fixed by #20006?

**Comment by MichaelLaufer** [comment]
@christian-heusel 
Can you provide the full spec that was concretized?
I believe that #20006 should solve both those issues. Let us know.

**Comment by christian-heusel** [comment]
@MichaelLaufer I just fixed it myself, but essentially my result is the same as the mentioned pull request so the problem is most likely fixed by it! 😄 

What still is unclear to me why in the mentioned commit  https://github.com/spack/spack/commit/42008e532f28c41d8e134052e1a109c65a8440f9 the maximum number of jobs is suddenly 10 instead of the 20 jobs that were possible before.


@adamjstewart @MichaelLaufer thank you both for your quick help!

**Comment by MichaelLaufer** [comment]
@christian-heusel 
Yes that is a valid comment:
20 is the limit imposed by WRF. @ptooley requested a change to a max of 10 a while back, presumably due to some bugs with high job count. Either way, the comment should surely match the code, so we will fix at the next PR.

**Comment by christian-heusel** [comment]
Thanks for clarifying! 😊 

You decide whether that should keep this issue open or not since my main issue was solved in #20006!
